components/BookSearch.tsx

```diff
@@ -1,125 +1,152 @@
-import React, { useState, useEffect } from 'react';
-import Book from './Book';
-import styles from '../styles/book-search.module.css';
+import React, { useEffect, useState } from "react";
 
-const baseUrl = 'https://gutendex.com/books';
-export const languages = ['pl', 'en', 'fr'];
+import styles from "../styles/book-search.module.css";
+import { Book } from "./Book";
+import type { AppContextType, Book_t, JsonResponse } from "./appContext";
 
+const baseUrl = "https://gutendex.com/books";
+export const languages = ["pl", "en", "fr"];
 
-export function toggleStringInList(string, list) {
+const DELAY = 300;
+
+export function toggleStringInList(string: string, list: string[]): string[] {
```

Comment:
nie potrzebujesz tu typu
```suggestion
export function toggleStringInList(string: string, list: string[]) {
```